internal/pkg/policy/policy_output_integration_test.go

```diff
@@ -202,3 +203,108 @@ func createAgent(ctx context.Context, t *testing.T, index string, bulker bulk.Bu
 
 	return agentID
 }
+
+func TestPolicyOutputESPrepareRemoteES(t *testing.T) {
+	ctx := testlog.SetLogger(t).WithContext(context.Background())
+	index, bulker := ftesting.SetupCleanIndex(ctx, t, dl.FleetAgents)
+
+	agentID := createAgent(ctx, t, index, bulker, map[string]*model.PolicyOutput{})
+	agent, err := dl.FindAgent(
+		ctx, bulker, dl.QueryAgentByID, dl.FieldID, agentID, dl.WithIndexName(index))
+	if err != nil {
+		require.NoError(t, err, "failed to find agent ID %q", agentID)
+	}
+
+	output := Output{
+		Type:         OutputTypeRemoteElasticsearch,
+		Name:         "test remote output",
+		ServiceToken: "token1",
+		Role: &RoleT{
+			Sha2: "new-hash",
+			Raw:  TestPayload,
+		},
+	}
+	policyMap := map[string]map[string]interface{}{
+		"test remote output": map[string]interface{}{
+			"hosts": []interface{}{"http://localhost:9200"},
+		},
+	}
+
+	err = output.prepareElasticsearch(
+		ctx, zerolog.Nop(), bulker, bulker, &agent, policyMap, false)
+	require.NoError(t, err)
+
+	// need to wait a bit before querying the agent again
+	// TODO: find a better way to query the updated agent
+	time.Sleep(time.Second)
+
+	got, err := dl.FindAgent(
+		ctx, bulker, dl.QueryAgentByID, dl.FieldID, agentID, dl.WithIndexName(index))
+	if err != nil {
+		require.NoError(t, err, "failed to find agent ID %q", agentID)
+	}
```

Comment:
Perhaps put this in a loop with some max retries before erroring the test out?

Reply:
updated